% Latex template: mahmoud.s.fahmy@students.kasralainy.edu.eg
% For more details: https://www.sharelatex.com/learn/Beamer

\documentclass{beamer}					% Document class
\usetheme{Dresden}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{tikz}
\usepackage{colortbl}
\usepackage{amsfonts,amsmath,amssymb}
\usepackage{enumerate}
\usepackage{enumitem}
\usepackage[french]{babel}
\usepackage{mathrsfs}
\usepackage{graphicx}
\usepackage{appendix}
\usepackage{float}
\usepackage{caption}
\usepackage{bbm}
\usepackage[singlespacing]{setspace}
\usepackage{float}
\usepackage{textcomp}
\usepackage{apptools}
\usepackage{tabularx}
\usepackage{makecell}
\usepackage{blkarray}
\usepackage{calc}
\usepackage{multirow}
\usepackage{array}
\usepackage{bigdelim}
\usetikzlibrary{calc,trees,positioning,arrows,chains,shapes.geometric,%
    decorations.pathreplacing,decorations.pathmorphing,shapes,%
    matrix,shapes}

\tikzstyle{decision} = [diamond, draw, text width=4.5em,
                        text badly centered, node distance=2cm,
                        inner sep=0pt]
\tikzstyle{block} = [rectangle, draw,
                     text centered, rounded corners,
                     minimum height=4em]
\tikzstyle{line} = [draw, -latex']
\tikzstyle{cloud} = [draw, circle, minimum height=3em]

%% Symboles
\newcommand{\R}{\mathbb{R}}

%% Commentaires
\newcommand{\MM}[1]{\textcolor{red}{#1}}


\tikzset{
>=stealth',
  decoration={brace},
  tuborg/.style={decorate},
  tubnode/.style={midway, right=2pt},
}



% Uncomment this to have the outline at the beginning of each section highlighted.
\AtBeginSection[]
{
 \begin{frame}{Plan}
   \tableofcontents[currentsection]
 \end{frame}
}

\mode<presentation>						% Set options
{
  \usetheme{default}					% Set theme
  \usecolortheme{default} 				% Set colors
  \usefonttheme{default}  				% Set font theme
  \setbeamertemplate{caption}[numbered]	% Set caption to be numbered
  \addtobeamertemplate{footline}{\insertframenumber/\inserttotalframenumber}
}

\newcolumntype{C}[1]{>{\centering\arraybackslash}m{#1}}
\newcommand{\pointcol}[1]{
\begin{tikzpicture}
\filldraw[fill=#1, draw=#1] circle (3pt);
\end{tikzpicture}
}

<<setup, include=FALSE, echo=FALSE>>=
library(knitr)
library(xtable)
library(doParallel)
knitr::opts_chunk$set(echo=FALSE, warning = FALSE, fig.path = "figure/", cache.path = "cache/", opts.label="kill_prefix")
knitr::opts_knit$set(root.dir = "..")
@


\title{Exploration des méthodes d'analyse de données compositionnelles pour l'étude du microbiome}
\author{Clément Hardy\\ encadré par Mahendra Mariadassou, Magali Berland }
\institute{Inra Jouy en Josas}
\date{29 Août 2018}
\titlegraphic{\includegraphics[width=1.5cm]{logo_paris_sud.jpg}\hspace*{4cm}~%
   \includegraphics[width=2.5cm]{../rapport/logoINRA2.jpg}
}

\begin{document}

<<chargement_source, echo=FALSE,  include=FALSE>>=
library(ellipse)
library(compositions)
source("Rscript/read_metagenomic_data.R")
source("Rscript/coda.R")
source("Rscript/comparaison_clustering.R")
source("Rscript/graph.R")
source("Rscript/bootstrap.R")
source("Rscript/test_bootstrap.R")
source("Rscript/tree_phyloseq.R")
source("Rscript/apprentissage_supervise.R")
source("Rscript/fonction_presentation.R")
@


<<chargement RData, echo=FALSE, include=FALSE>>=
load("rapport/resultats.RData")
@

\begin{frame}
  \titlepage
\end{frame}

% Outline
% This page includes the outline (Table of content) of the presentation. All sections and subsections will appear in the outline by default.
\begin{frame}<beamer>{Plan}
\tableofcontents
\end{frame}

% The following is the most frequently used slide types in beamer
% The slide structure is as follows:
%
%\begin{frame}{<slide-title>}
%	<content>
%\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section[Contexte]{Contexte du Stage}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}{Introduction}
Métagénomique: méthode d'étude du contenu génétique d'échantillons issus d'environnements complexes

\begin{center}
\includegraphics[scale=0.3]{shotgun_sequencing.png}
\end{center}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Présentation des données}
\begin{small}
\input{tikz_figures/otu_table}
\end{small}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Problématique et Objectifs du stage}
\(
\left.
\begin{tabular}{l}
 (1) Comprendre les donneés compositionnelles \\
 (2) Appliquer les méthodes compositionnelles \\
 (3) Analyse multivariées \\
 (4)  Apprentissage de densité \\
\end{tabular}
\right\rbrace
\Rightarrow \text{ Simulateur}\) 
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section[Compositionnalité]{Données compositionnelles}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Définition}
	Composition: \[
	x=\left[ x_1,x_2,...,x_D\right],\ x_i>0
	\]
	ne contient que de l'information \textbf{relative} (ex: pourcentage d'une masse).\\
	\vspace{0.5cm}
	Simplexe:
	\[
    S^D=\lbrace x=\left[x_1, x_2,...,x_D\right]|x_i>0,i=1,2,...,D;\sum_{i=1}^Dx_i=1\rbrace
    \]
\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
<<fig.height=4>>=
par(mar=rep(0,4))
ternary_diagram2()
par(mar=rep(4,4))
@
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]{Espace des compositions}

\begin{center}

<<fig.height=3>>=
c <- matrix(c(c(1,2,7)/10, ilr_inverse(c(-1.2,0)), c(1,8,1, 10/3, 10/3, 10/3)/10), byrow=TRUE, nrow=4)
l <- ligne(1.2,c(0.2,1,3))
y <- seq(0, 2*pi, length=1000)
circl <- (cbind(1*cos(y)-1.2, 1*sin(y))) %>% ilr_inverse()

ternary_diagram_vide()
c[1:3,] %>% coord_ternary_diagram() %>% points(col=c(11,2,4), pch=16, cex=1.5)

@

\begin{tabular}{|c|c|c|c|}
\hline
& x1(A) & x2(B) & x3(C) \\
\hline
\pointcol{green} & \Sexpr{c[1,1] %>% round(2)} & \Sexpr{c[1,2] %>% round(2)} & \Sexpr{c[1,3]} \\
\hline
\pointcol{red} & \Sexpr{c[2,1] %>% round(2)} & \Sexpr{c[2,2] %>% round(2)} & \Sexpr{c[2,3] %>% round(2)} \\
\hline
\pointcol{blue} & \Sexpr{c[3,1] %>% round(2)} & \Sexpr{c[3,2] %>% round(2)} & \Sexpr{c[3,3] %>% round(2)} \\
\hline
\end{tabular}
\end{center}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Opérations de base}
\begin{small}
		Closure:
		\[
			C(x)=\left[\frac{ x_1}{\sum_{i=1}^Dx_i},				\frac{x_2}{\sum_{i=1}^Dx_i},...,						\frac{ x_D}{\sum_{i=1}^Dx_i}     \right]
		\]
        Perturbation(\(\sim +\)) :
        \[
        x,y\in S^D,\ x\oplus y=C\left[x_1y_1,x_2y_2,...,x_Dy_D\right]
        \]
        Puissance(\(\sim \times\)) :
        \[
        x\in S^D,\ \alpha\in \mathbb{R},\ \alpha\odot  x=C\left[x_1^{\alpha},x_2^{\alpha},...,x_D^{\alpha}\right]
        \]
        Produit scalaire :
        \[
        x,y\in S^D,\ \left\langle x,y\right\rangle_a=\frac{1}{2D}\sum_{i=1}^D\sum_{j=1}^Dln
        \frac{x_i}{x_j}ln\frac{y_i}{y_j}
        \]
\end{small}

\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]{Objets géométrique}
\begin{center}
\(t=t_0\oplus (\alpha \odot t_1)\)

<<fig.height=3>>=
c <- matrix(c(c(3.5,4,3)/10.5, ilr_inverse(c(-1.2,0)), c(1,8,1, 10/3, 10/3, 10/3)/10), byrow=TRUE, nrow=4)

t <- c(2,1,3)/6
ternary_diagram_vide()
points(c[1,] %>% coord_ternary_diagram(), pch=16, col=4, cex=2)
points(c[2,] %>% coord_ternary_diagram(), pch=16, col='red',cex=2)
points(c[3,] %>% coord_ternary_diagram(), pch=16,cex=2)

l <- ligne(c[3,],t)
l %>% coord_ternary_diagram() %>% lines(col=15, lwd=2)

l <- ligne(c[2,],t)
l %>% coord_ternary_diagram() %>% lines(col=15, lwd=2)

l <- ligne(c[1,],t)
l %>% coord_ternary_diagram() %>% lines(col=15, lwd=2)


t1 <- c(1,4,3)/8

l <- ligne(c[3,],t1)
l %>% coord_ternary_diagram() %>% lines(col=3, lwd=2)

l <- ligne(c[2,],t1)
l %>% coord_ternary_diagram() %>% lines(col=3, lwd=2)

l <- ligne(c[1,],t1)
l %>% coord_ternary_diagram() %>% lines(col=3, lwd=2)
@

\begin{tabular}{cc}
\begin{tabular}{|c|c|c|c|}
\hline
& x1(A) & x2(B) & x3(C) \\
\hline
\pointcol{blue} & \Sexpr{c[1,1] %>% round(2)} & \Sexpr{c[1,2] %>% round(2)} & \Sexpr{c[1,3] %>% round(2)} \\
\hline
\pointcol{red} & \Sexpr{c[2,1] %>% round(2)} & \Sexpr{c[2,2] %>% round(2)} & \Sexpr{c[2,3] %>% round(2)} \\
\hline
\pointcol{black} & \Sexpr{c[3,1] %>% round(2)} & \Sexpr{c[3,2] %>% round(2)} & \Sexpr{c[3,3] %>% round(2)} \\
\hline
\end{tabular}

&

\begin{tabular}{|c|c|c|c|}
\hline
& x1 & x2 & x3 \\
\hline
\pointcol{yellow} & \Sexpr{t[1] %>% round(2)} & \Sexpr{t[2] %>% round(2)} & \Sexpr{t[3] %>% round(2)} \\
\hline
\pointcol{green} & \Sexpr{t1[1] %>% round(2)} & \Sexpr{t1[2] %>% round(2)} & \Sexpr{t1[3] %>% round(2)} \\
\hline
\end{tabular}
\end{tabular}
\end{center}
\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]
<<>>=
mat1 <- matrix(c(0.3,0,0,0.3), nrow=2) *0.1
mat2 <- matrix(c(0.4,-0.5,-0.5,2), nrow=2)*0.1
mat3 <- matrix(c(0.8,0.9,0.9,1.5), nrow=2)*0.1
u1 <- c(0,0)*0.5
u2 <- c(3,-2)*0.5
u3 <- c(-1.8,-0.5)*0.5
col1="red"
col2="green"
col3="gray"
@


<<fig.height=4>>=
          ternary_diagram_vide()
ellipse(mat2, centre = u2) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col2)
ellipse(mat2, centre = u2, level=0.8) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col2)
ellipse(mat2, centre = u2, level=0.4) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col2)


ellipse(mat1, centre = u1) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col1)
ellipse(mat1, centre = u1, level=0.8) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col1)
ellipse(mat1, centre = u1, level=0.4) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col1)


ellipse(mat3, centre = u3) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col3)
ellipse(mat3, centre = u3, level=0.8) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col3)
ellipse(mat3, centre = u3, level=0.4) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col3)

              @
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Transformations}
Plusieurs transformations permettent de passer de \(S^D\) à \(\R^D\) ou \(\R^{D-1}\)
\begin{itemize}
	\item\(
	alr:S^D\rightarrow \R^{D-1},\ alr\left(x\right)=\left[ln\left(\frac{x_1}{x_D}\right), ln\left(\frac{x_2}{x_D}\right),...,ln\left(\frac{x_{D-1}}{x_D}\right)\right]
	\)
	\pause
	\item\(
    clr:S^D\rightarrow \R^{D},\ clr\left(x\right)=\left[ln\frac{x_1}{g(x)},...,ln\frac{x_D}{g(x)}\right]
    \)\\
    avec \(g(x)\) le centre de la composition défini par:
    \[
    g(x)=\left(\prod^D_{i=1}x_i\right)^{1/D}
    \]
   \pause
    \item  \(
    ilr:S^D\rightarrow \R^{D-1},\ ilr\left(x\right)=clr\left(x\right)\Psi^{'}\)

    \[
    \text{ avec }\Psi  \text{ l'image(clr) des vecteurs de la base.}
    \]
\end{itemize}
\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]{Aitchison geometry}

\begin{minipage}{\linewidth}
      \centering
      \begin{minipage}{0.65\linewidth}
<<fig.height=4>>=
par(mar=c(rep(0,4)))
c <- matrix(c(c(1,2,7)/10, ilr_inverse(c(-1.2,0)), c(1,8,1, 10/3, 10/3, 10/3)/10), byrow=TRUE, nrow=4)
y <- seq(0, 2*pi, length=1000)
circl <- (cbind(1*cos(y)-1.2, 1*sin(y))) %>% ilr_inverse()

ternary_diagram1(c, colour=c(11, 2, 4, 1), style=c(16, 16, 16, 3),add_line = l, colour_line = 15,
                add_circle = circl, colour_circle = 14,
                asp = 1, axes = FALSE)
@
\end{minipage}
\hspace{0.05\linewidth}
\begin{minipage}{0.25\linewidth}
\resizebox{\linewidth}{!}{%
 \begin{tabular}{|c|c|c|c|}
\hline
& x1 & x2 & x3 \\
\hline
\pointcol{green} & \Sexpr{c[1,1] %>% round(2)} & \Sexpr{c[1,2] %>% round(2)} & \Sexpr{c[1,3]} \\
\hline
\pointcol{red} & \Sexpr{c[2,1] %>% round(2)} & \Sexpr{c[2,2] %>% round(2)} & \Sexpr{c[2,3] %>% round(2)} \\
\hline
\pointcol{blue} & \Sexpr{c[3,1] %>% round(2)} & \Sexpr{c[3,2] %>% round(2)} & \Sexpr{c[3,3] %>% round(2)} \\
\hline
\end{tabular}
}
\end{minipage}
\end{minipage}

\begin{minipage}{\linewidth}
      \centering
      \begin{minipage}{0.65\linewidth}
<<fig.height=4>>=
l <- ligne(c[3,],c(1.2,1,3))
c_ilr <- ilr(c)
l_ilr <- ilr(l)
circl_ilr <- ilr(circl)
plot(c_ilr, col=c(11, 2, 4, 1), pch=c(16, 16, 16, 3), cex=1.5, xlim = c(-2.2,2), ylim=c(-1,1.5), asp=1, xlab = "", ylab="")
lines(l_ilr, col=15, lwd=3)
lines(circl_ilr, col=14, lwd=2)
@
\end{minipage}
\hspace{0.05\linewidth}
\begin{minipage}{0.25\linewidth}
\resizebox{\linewidth}{!}{%
 \begin{tabular}{|c|c|c|}
\hline
& x1 & x2  \\
\hline
\pointcol{green} & \Sexpr{c_ilr[1,1] %>% round(2)} & \Sexpr{c_ilr[1,2] %>% round(2)} \\
\hline
\pointcol{red} & \Sexpr{c_ilr[2,1] %>% round(2)} & \Sexpr{c_ilr[2,2] %>% round(2)} \\
\hline
\pointcol{blue} & \Sexpr{c_ilr[3,1] %>% round(2)} & \Sexpr{c_ilr[3,2] %>% round(2)}  \\
\hline
\end{tabular}
}
\end{minipage}
\end{minipage}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Transformations inverses}
\begin{itemize}
\item \(
clr^{-1}\left(\xi\right)=C\left( \exp\left(\xi_1\right),\ ...,\  \exp\left(\xi_D\right)\ \right)
\)

\item \(
ilr^{-1}\left(\xi\right)=C\left(\exp{\xi\Psi}\right)
\)
\end{itemize}


\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]{Morphisme}

<<>>=
x <- runif(3, 0.1,1.2) %>% round(digits = 2) %>% norm_data()
x1 <- runif(3, 0.1,1.2) %>% round(digits = 2) %>% norm_data()
@

\begin{center}
\begin{tabular}{cc}
x=[\Sexpr{x}] & x1=[\Sexpr{x1}]
\end{tabular}
\end{center}


\begin{tikzpicture}[node distance=4cm, auto]
\node[] (X) {\begin{tabular}{c} x \\ x1 \end{tabular}};
\node[right of=X] (X1) {[\Sexpr{perturbation(x,x1) %>% round(digits=2)}]};
\node[right of=X1] (X2) {[\Sexpr{perturbation(x,x1) %>% ilr %>% round(digits=2)}]};
\node[below of=X, node distance=2cm] (X_1) {\begin{tabular}{c} x \\ x1 \end{tabular}};
\node[right of=X_1] (X3) {\begin{tabular}{c}[\Sexpr{x %>% ilr() %>% round(digits=2)}] \\ 
 \([\Sexpr{x1 %>% ilr() %>% round(digits=2)}] \)
\\
\end{tabular}};
\node[right of=X3] (X4) {[\Sexpr{((x %>% ilr) + (x1%>% ilr)) %>% round(digits=2)}]};


\path[line] (X) --  node[midway, above] {perturbation} (X1);
\path[line] (X1) --  node[midway, above] {\(ilr\)} (X2);
\path[line] (X_1) --  node[midway, above] {\(ilr\)} (X3);
\path[line] (X3) --  node[midway, above] {somme} (X4);
\end{tikzpicture}
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}
La transformation ilr permet de repasser dans un espace usuel.
\\
\begin{itemize}
\item analyse multivariée
\item test de normalité
\item clustering
\item estimation de densité
\item ...
\end{itemize}


\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]

<<>>=
mat1 <- matrix(c(0.3,0,0,0.3), nrow=2) *0.1
mat2 <- matrix(c(0.4,-0.5,-0.5,2), nrow=2)*0.1
mat3 <- matrix(c(0.8,0.9,0.9,1.5), nrow=2)*0.1
u1 <- c(0,0)*0.5
u2 <- c(3,-2)*0.5
u3 <- c(-1.8,-0.5)*0.5
col1="red"
col2="green"
col3="gray"
@

\begin{minipage}{\linewidth}
      \centering
      \begin{minipage}{0.45\linewidth}
          <<fig.height=6.5>>=
          		
ellipse(mat2, centre = u2) %>% plot(type="l", asp=1, xlim=c(-2,2), ylim=c(-2,1), col=col2)
ellipse(mat2, centre = u2, level=0.8) %>% lines(col=col2)
ellipse(mat2, centre = u2, level=0.4) %>% lines(col=col2)

ellipse(mat1, centre =u1) %>% lines(col=col1)
ellipse(mat1, centre =u1, level=0.8) %>% lines(col=col1)
ellipse(mat1, centre =u1, level=0.4) %>% lines(col=col1)

ellipse(mat3, centre =u3) %>% lines(col=col3)
ellipse(mat3, centre =u3, level=0.8) %>% lines(col=col3)
ellipse(mat3, centre =u3, level=0.4) %>% lines(col=col3)

              @
      \end{minipage}
      \hspace{0.05\linewidth}
      \begin{minipage}{0.45\linewidth}
          <<fig.height=6.5>>=
          ternary_diagram_vide()
ellipse(mat2, centre = u2) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col2)
ellipse(mat2, centre = u2, level=0.8) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col2)
ellipse(mat2, centre = u2, level=0.4) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col2)


ellipse(mat1, centre = u1) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col1)
ellipse(mat1, centre = u1, level=0.8) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col1)
ellipse(mat1, centre = u1, level=0.4) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col1)


ellipse(mat3, centre = u3) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col3)
ellipse(mat3, centre = u3, level=0.8) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col3)
ellipse(mat3, centre = u3, level=0.4) %>% ilr_inverse %>% coord_ternary_diagram() %>% lines(col=col3)

              @
      \end{minipage}
  \end{minipage}

\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section[Application]{Application en métagénomique}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Jeux de données}
Jeux de données de comptage \(\Rightarrow\) prétraitement nécessaire (présence de 0).\newline
\\
estimateur maximum a posteriori (MAP) sous un a priori \(\mathcal{D}(1,1)\):
\[
MAP(p)=\underset{p}{argmax}\left(p|x\right)=\frac{x_i+1}{\sum_{i=1}^D\left(x_i+1\right)}
\]

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Différence entre MAP et maximun de vraisemblance}

\begin{center}
\begin{tabular}{|c|c|c|c|c|c|}
\hline
\multicolumn{6}{|c|}{Comptages} \\
\hline
2 & 5 & 12 & 0 & 1 & 0 \\
\hline
\end{tabular}
\end{center}

\vspace{1.5cm}

\begin{tabular}{cc}

\setlength{\extrarowheight}{4pt}
\begin{tabular}{|c|c|c|c|c|c|}
\hline
\multicolumn{6}{|c|}{Compositions(MV)} \\
\hline
\(\frac{2}{20}\) & \(\frac{5}{20}\) & \(\frac{12}{20}\) & 0 & \(\frac{1}{20}\) & 0 \\
\hline
\end{tabular}

&

\setlength{\extrarowheight}{4pt}
\begin{tabular}{|c|c|c|c|c|c|}
\hline
\multicolumn{6}{|c|}{Compositions(MAP)} \\
\hline
\(\frac{3}{26}\) & \(\frac{3}{13}\) & \(\frac{1}{2}\) & \(\frac{1}{26}\) & \(\frac{1}{13}\) & \(\frac{1}{26}\) \\
\hline
\end{tabular}


\end{tabular}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Liver\cite{Liver}}
\resizebox{\linewidth}{!}{%
<<>>=
head(metadata_liver) %>% kable()
@
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Réduction de dimension}
<<graph biplot Liver, fig.height = 4.5>>=
graph_biplot_normale(data=liver_500, metadata_group=metadata_liver$status, nb_graph=1, title="liver", legend_title = "status")[[1]]

@
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[fragile]{Classification k-means}
\resizebox{\linewidth}{!}{%
\begin{tabular}{cc}
comptage & ilr \\
<<>>=
kable(k_liver$comptage_table)
@
&
<<>>=
kable(k_liver$ilr_table)
@
\end{tabular}
}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section[Simulateur]{Simulateur de données}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Objectif}
Construire un simulateur qui génère des données réalistes(bootstrap paramétrique)\\

\begin{itemize}
\item Validation de méthode 
\item Benchmark
\end{itemize}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Méthodes de simulation}
\[
X \overset{MAP}{\rightarrow} \left(S^p\right)^n \overset{ilr}{\rightarrow}    \left(\mathbb{R}^{p-1}\right)^n \overset{\overset{\text{réduction de}} {dimension}}{\rightarrow} \left(\mathbb{R}^D\right)^n \rightarrow \text{Apprentissage densité}
\]
\[
\text{densité} \overset{simulation}{\rightarrow} \left(\mathbb{R}^D\right)^n \rightarrow
\left(\mathbb{R}^{p-1}\right)^n \overset{ilr^{-1}}{\rightarrow}\left(S^{p-1}\right)^n  \overset{multinomiale/\overset{zero}{inflation}}{\rightarrow} \tilde{X}
\]
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Simulations de données ("bootstrap")}{Réduction de dimension}

ACP Probabiliste:
\[
Y_i=Wx_i+\mu+\epsilon_i, \ x_i\sim \mathcal{N}\left(0,\mathbb{I}_d\right), \ \epsilon_i \sim \mathcal{N}\left(0, \sigma^2\mathbb{I}_p\right)
\]
\newline
Choix de la dimension fait en regardant l'heuristique de pente.
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Choix de la dimension}
<<fig.height=4.5>>=
y <- matrix(0, ncol=10, nrow=1000)
y[,1:3] <- mvrnorm(n=1000, rep(0, 3), matrix(c(1,0,0,0,1,0,0,0,1),nrow=3)) 
y <- y + mvrnorm(1000, rep(0, 10), seq(2,0.1,length=10)^7*diag(10))
z <- cov(y) %>% eigen()
plot(z$values, ylab="y", xlab="Index", pch=16)
abline(v=4, col='red')
@
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
<<fig.height=3.5>>=
g <- graph_biplot_normale(liver_500, "data", legend=FALSE)
g[[1]] + geom_density2d(aes(X1, X2))
@

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Simulations de données ("bootstrap paramétrique")}{Estimation loi}
Mélange Gaussien:

\[
g\left(x, \Theta\right)=\sum_{k=1}^K \pi_kf(x,\theta_k)
\]
avec \(f\) la fonction de densité de la loi normale multivariée.

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{mélange gaussien}
<<fig.height=4>>=
x <- seq(-5, 6, length=10000)
y1 <- dnorm(x, mean=-1.8)*0.3
y2 <- dnorm(x, mean=1.6, sd=1.4)*0.5
y3 <- dnorm(x, mean=-3, sd=0.4)*0.2
n <- 1000000
r1 <- rnorm(n*0.3, -1.8)
r2 <- rnorm(n*0.5, 1.6, 1.4)
r3 <- rnorm(n*0.2, -3, 0.4)

hist(c(r1, r2, r3), proba=TRUE, breaks = 50, main="melange gaussien", xlab = "x")
lines(x, y1, col="red")
lines(x, y2, col='blue')
lines(x,y3, col="green")
lines(x, y1+y2+y3)
legend("topright", lty=c(1,1,1), col=c("red", "blue", "green", "black"), legend=c("G1","G2", "G3","G1+G2+G3"))

@
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]{Première évaluation}

\begin{center}
<<boot6, fig.height=3.5, cache=TRUE>>=
liver_500_boot <- bootstrap(liver_500, nb_cluster = 13, nb_axe = 17, type="comptage", zero_inflated = FALSE)
data <- rbind(liver_500, liver_500_boot$data)
metadata <- c(rep("real", nrow(liver_500)), rep("simu", nrow(liver_500_boot$data))) %>% as.factor()
graph_biplot_normale(data, metadata, 1, title = "Liver", legend=FALSE, legend_title = "data")[[1]]
@

\begin{tabular}{cc}
réelles & \pointcol{red} \\
simulées & \pointcol{blue}
\end{tabular}
\end{center}
\end{frame}



\begin{frame}{Classificateur}

\begin{tikzpicture}[node distance=2cm, auto]
\node[cloud] (Donne_reel) {réels};
\node[below of=Donne_reel] (temp) {};
\node[cloud, below of=temp] (Donne_simu) {simu};
\node[right of=temp, node distance=2cm] (classificateur) {Classificateur};
\node[right of=classificateur] (temp2) {};
\node[cloud, above of=temp2] (Donne_reel2) {réels};
\node[cloud, below of=temp2] (Donne_simu2) {simu};
\node[right of=temp2, node distance=4cm] (eval)  {\begin{tabular}{|c|c|c|}
\hline
& réelles & simulées \\
\hline
réelles(pred) & \Sexpr{t_liver_500_sans$all$random_forest[1,1]} & \Sexpr{t_liver_500_sans$all$random_forest[1,2]} \\
\hline
simulées(pred) & \Sexpr{t_liver_500_sans$all$random_forest[2,1]}   & \Sexpr{t_liver_500_sans$all$random_forest[2,2]}\\
\hline
\end{tabular}};

\path[line] (Donne_reel) --  (classificateur);
\path[line] (Donne_simu) --  (classificateur);
\path[line] (classificateur) --  (eval);
\path[line] (Donne_reel2) --  (eval);
\path[line] (Donne_simu2) --  (eval);
\end{tikzpicture}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}

<<boot1, fig.height=4, cache=TRUE>>=
liver_500_boot <- bootstrap(liver_500, nb_cluster = 13, nb_axe = 17, zero_inflated = FALSE)
data <- rbind(liver_500, liver_500_boot$data)
metadata <- c(rep("real", nrow(liver)), rep("simu", nrow(liver_boot$data))) %>% as.factor()
d <- data.frame(data, metadata=metadata)
ggplot(d) + geom_histogram(aes(CAG00214_hs_9.9, fill=metadata), position="dodge", binwidth = 1) + xlim(c(-1, 30))
@


\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}{Zero inflation}
Zero inflation: mettre des zeros aléatoirement dans le jeu de donnée simulé.
<<>>=
set.seed(2)
l1 <-((runif(7,1,10)*c(1,0,1,0,0,1,1)) %>% round(digits=0))
l2 <- ((runif(7,1,10)*c(0,1,0,0,0,1,0)) %>% round(digits=0))
@
\vspace*{0.3cm}
\begin{tabular}{cc}
\begin{tikzpicture}
\draw [fill=red,draw=none] (0,0) rectangle (0.3,0.3);
\end{tikzpicture}
&
Gaussienne 1 \\

\begin{tikzpicture}
\draw [fill=green,draw=none] (0,0) rectangle (0.3,0.3);
\end{tikzpicture}
&
Gaussienne 2 \\
\end{tabular}

\begin{tabular}{ccc}
\begin{tabular}{|>{\columncolor{red}}c|>{\columncolor{red}}c|>{\columncolor{red}}c|>{\columncolor{green}}c|>{\columncolor{green}}c|>{\columncolor{green}}c|>{\columncolor{green}}c}
\multicolumn{6}{c}{Compositions} \\
\Sexpr{l1[1]} & \Sexpr{l1[2]} & \Sexpr{l1[3]} & \Sexpr{l1[4]} & \Sexpr{l1[5]} & \Sexpr{l1[6]} & \Sexpr{l1[7]} \\

\Sexpr{l2[1]} & \Sexpr{l2[2]} & \Sexpr{l2[3]} & \Sexpr{l2[4]} & \Sexpr{l2[5]} & \Sexpr{l2[6]} & \Sexpr{l2[7]} \\
\end{tabular}

&
\(\rightarrow\)
&
\begin{tabular}{|>{\columncolor{red}}c|>{\columncolor{green}}c}
\multicolumn{2}{c}{}\\
  1/3 & 1/2 \\
 2/3 & 3/4
\end{tabular}
\end{tabular}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{frame}
  \[
  \begin{array}{c|>{\columncolor{red}}c|>{\columncolor{green}}c|}
 \multicolumn{3}{c}{\quad \quad \quad \quad \ \ Gaussienne} \\ 
 \multicolumn{3}{c}{\quad \quad \quad \quad \ \ $\downbracefill$} \\
  \cline{2-3}
  \ldelim \{{2}{*}[OTUS]& 
  1/3 & 1/2  \\
  \cline{2-3}
   & 2/3 & 3/4 \\
   \cline{2-3}
  \end{array}
\]

\begin{center}
\begin{tabular}{cc}
\begin{tabular}{|>{\columncolor{red}}c|>{\columncolor{red}}c|>{\columncolor{red}}c|>{\columncolor{green}}c|>{\columncolor{green}}c|>{\columncolor{green}}c|>{\columncolor{green}}c|>{\columncolor{green}}c|}
\multicolumn{8}{c}{Avant zero inflation}\\
1 & 2 & 18 & 5 & 9 & 7 & 0 & 21 \\
32 & 2 & 3 & 6 & 2 & 1 & 2 & 14
\end{tabular}
 
 &
 
 \begin{tabular}{|>{\columncolor{red}}c|>{\columncolor{red}}c|>{\columncolor{red}}c|>{\columncolor{green}}c|>{\columncolor{green}}c|>{\columncolor{green}}c<|>{\columncolor{green}}c|>{\columncolor{green}}c|}
 \multicolumn{8}{c}{Après zero inflation}\\
1 & 0 & 18 & 0 & 9 & 0 & 0 & 21 \\
0 & 0 & 3 & 0 & 0 & 1 & 0 & 14
\end{tabular}
 \end{tabular}
 \end{center}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}

<<boot2, fig.height=4, cache=TRUE>>=
liver_500_boot <- bootstrap(liver_500, nb_cluster = 13, nb_axe = 17, zero_inflated = TRUE)
data <- rbind(liver_500, liver_500_boot$data)
metadata <- c(rep("real", nrow(liver)), rep("simu", nrow(liver_boot$data))) %>% as.factor()
d <- data.frame(data, metadata=metadata)
ggplot(d) + geom_histogram(aes(CAG00214_hs_9.9, fill=metadata), position="dodge", binwidth = 1)
@

\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
\begin{center}
<<boot3, fig.height=3.5,cache=TRUE>>=
liver_boot <- bootstrap(liver_500, nb_cluster = 13, nb_axe = 17, zero_inflated = TRUE)
data <- rbind(liver_500, liver_boot$data)
metadata <- c(rep("real", nrow(liver)), rep("simu", nrow(liver_boot$data))) %>% as.factor()
graph_biplot_normale(data, metadata, 1, "liver", "data")[[1]]

@

\begin{tabular}{cc}
réelles & \pointcol{red} \\
simulées & \pointcol{blue}
\end{tabular}
\end{center}
\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Résultat après zero inflated}
\begin{center}
\begin{tabular}{|c|c|c|}
\hline
& réelles & simulées \\
\hline
réelles(pred) & \Sexpr{t_liver_500$all$random_forest[1,1]} & \Sexpr{t_liver_500$all$random_forest[1,2]} \\
\hline
simulées(pred) & \Sexpr{t_liver_500$all$random_forest[2,1]}   & \Sexpr{t_liver_500$all$random_forest[2,2]} \\
\hline
\end{tabular}
\end{center}
\vspace{1cm}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame} [fragile]{Problématique profondeur de séquencage}

<<boot4, fig.height=4.5,cache=TRUE>>=
liver_500_boot <- bootstrap_presentation(liver_500, nb_cluster = 13, nb_axe = 17, type="comptage", zero_inflated=FALSE)
data <- data.frame(data=c(liver_500_boot$data %>% apply(1, sum), liver_500 %>% apply(1, sum)), metadata=c(rep("simu",nrow(liver_500)),rep("real", nrow(liver_500))))
ggplot(data, aes(x=data, color=metadata, fill=metadata)) + geom_histogram(position="dodge", bins=30) +
labs(y="nombre", x="profondeur")
@


\end{frame}

\begin{frame}[fragile]
<<>>=
pro1 <- liver_500_boot$data %>% apply(1,sum) %>% mean
pro2 <- liver_500 %>% apply(1,sum) %>% mean
@
\begin{center}
\begin{tabular}{c|c}
simulées & réelles \\
\hline
\Sexpr{pro1}
&
\Sexpr{pro2}
\end{tabular}
\end{center}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section[Conclusion]{Perspectives et Conclusions}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Conclusion}
\begin{itemize}[label={}]
\item \textbf{Résultats}
\begin{itemize}[label={$\bullet$}]
\item Transformation log ratio accès à l'ensemble méthodes \\
d'analyses multivariée.
\item Gain au niveau de l'analyse des échantillons
\item Perte d'interprétabilité au niveau des OTU (espèces)
\end{itemize}
\item \textbf{Améliorations}
\begin{itemize}[label={$\bullet$}]
\item Zero inflation multivariée
\item Méthode d'apprentissage de la densité plus sophistiqué.
\end{itemize}
\end{itemize}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Annexes}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Annexe}{PPCA}
<<fig.height=4>>=

set.seed(20160919)
x <- rnorm(100)
noise <- matrix(rnorm(200, 0.1), ncol = 2)
mu <- c(-1, 1)
data.perfect <- (x %*% t(c(x = 1, y = 2)) + matrix(rep(mu, each =
                                                         length(x)), ncol = 2)) %>% as.data.frame()
segment.data <- cbind(data.perfect, data.perfect + noise)
names(segment.data) <- c("x", "y", "xend", "yend")
p.template <- ggplot(data = NULL, aes(x, y)) + geom_abline(slope = 2,
                                                           intercept = 3, color = "darkred") + scale_x_continuous(limits = c(-5,
                                                                                                                             3)) + scale_y_continuous(limits = c(-4, 5))
grid.arrange(ggplot(data.frame(x = x, y = 0), aes(x, y)) + geom_point()
             + geom_hline(yintercept = 0, color = "darkred") +
               ggtitle(expression(Latent~Space~(W))),
             p.template + geom_point(data = data.perfect) +
               ggtitle(expression(Parameter~Space~(mu+WB))),
             p.template + geom_point(data = data.perfect, color =
                                       "red") + ggtitle(expression(Observation~Space~(mu+WB))),
             p.template + geom_point(data = data.perfect, color = "red") +
               geom_point(data = data.perfect + noise) +
               geom_segment(data = segment.data, aes(xend = xend, yend
                                                     = yend), color = "black", alpha = 0.2) +
               ggtitle(expression(Observation~Space~(mu+WB+E))),
             ncol = 2)

@
\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Annexe}{conservation distance}

\resizebox{\textwidth}{!}{%
\begin{tabular}{ccc}
simplexe & alr & clr,ilr \\
\begin{tabular}{cccc}
& 1 & 2 & 3 \\
2 & 2.14 & & \\
3 & 2.37 & 2.8 & \\
4 & 1.39 & 1.2 & 1.7 \\
\end{tabular}
&
\begin{tabular}{cccc}
& 1 & 2 & 3 \\
2 & 2.82 & & \\
3 & 3.86 & 3.05 & \\
4 & 2.31 & 1.2 & 2.08 \\
\end{tabular}
&
\begin{tabular}{cccc}
& 1 & 2 & 3 \\
2 & 2.14 & & \\
3 & 2.37 & 2.8 & \\
4 & 1.39 & 1.2 & 1.7 \\
\end{tabular}
\end{tabular}
}

\end{frame}


\begin{frame}{Annexe}{Transformation ilr}
Matrice utilisé pour coder la base:\\

\[
\begin{blockarray}{cccc}
x1 & x2 & x3 & x4  \\
\begin{block}{(cccc)}
  1 & 1 & 1 & -1 \\
1 & 1 & -1 & 0 \\
1 & -1 & 0 & 0 \\
\end{block}
\end{blockarray}
\]


L'image \(clr\) de la base
\[
\Psi =\begin{pmatrix}
\frac{1}{\sqrt{12}} & \frac{1}{\sqrt{12}} & \frac{1}{\sqrt{12}} & -\frac{3}{\sqrt{12}} \\
\frac{1}{\sqrt{6}} & \frac{1}{\sqrt{6}} & -\frac{2}{\sqrt{6}} & 0 \\
\frac{1}{\sqrt{2}} & -\frac{1}{\sqrt{2}} & 0 & 0
\end{pmatrix}
\]

\end{frame}


\begin{frame}{Annexe}{Transformation ilr}
Systeme générateur: \(e_i=C(1,1,....e,...,1)\) \\
On omet le dernier vecteur puis on orthogonalise avec gram-schmidt.
\[
wi=e_i-\sum_{k=1}^{i-1}\frac{\left\langle e_i, e_{k}\right\rangle}{||e_{k}||^2}e_{k}
\]
\end{frame}





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]{Annexe}{Correction profondeur de séquencage}
estimation nombre de zéro ajouté:
\[
\hat{p_i}=\sum_{j=1}^D \pi_{jk}*\tilde{S_i}
\]
avec \(\pi_{jk}\) la probabilité de mettre un zéro\\
\(\tilde{S_i}\) une composition simulée

\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}[fragile]{Annexe}{Correction profondeur de séquencage}
  \begin{tabular}{lr}
  &
  \(
  \begin{array}{c|>{\columncolor{red}}c|>{\columncolor{green}}c|}
 \multicolumn{3}{c}{\quad \quad \quad \quad \ \ Gaussienne} \\ 
 \multicolumn{3}{c}{\quad \quad \quad \quad \ \ $\downbracefill$} \\
  \cline{2-3}
  \ldelim \{{4}{*}[OTUS]& 
  1/3 & 1/2  \\
  \cline{2-3}
   & 2/3 & 3/4 \\
   \cline{2-3}
   & 1/5 & 4/5 \\
   \cline{2-3}
   & 2/9 & 2/5
  \end{array}
\)
\\

\begin{tabular}{>{\columncolor{red}}c|>{\columncolor{green}}c|}
\multicolumn{2}{c}{Compositions}\\
1/4 & 1/6 \\
2/7 & 4/13 \\
3/7 & 17/156 \\
1/28 & 5/12 \\
\end{tabular}
&
\begin{tabular}{>{\columncolor{red}}c|>{\columncolor{green}}c|}
\Sexpr{((1/4)*(1/3)+(2/7)*(2/3)+(3/7)*(1/5)+(1/28)*(2/9)) %>% round(digits = 2)} & \Sexpr{((1/6)*(1/2)+(4/13)*(3/4)+(17/156)*(4/5)+(5/12)*(2/5)) %>% round(digits = 2)}
\end{tabular}
\end{tabular}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Annexe}{Problématique profondeur de séquencage}

<<boot5, fig.height=4.5, cache=TRUE>>=
liver_500_boot <- bootstrap(liver_500, nb_cluster = 13, nb_axe = 17, type="comptage")
data <- data.frame(data=c(liver_500_boot$data %>% apply(1, sum), liver_500 %>% apply(1, sum)), metadata=c(rep("simu",nrow(liver_500)),rep("real", nrow(liver_500))))
ggplot(data, aes(x=data, color=metadata, fill=metadata)) + geom_histogram(position="dodge", bins=30) +
labs(y="nombre", x="profondeur")
@

\end{frame}


\begin{frame}[fragile]{Annexe}{Problématique profondeur de séquencage}
<<>>=
pro1 <- liver_500_boot$data %>% apply(1,sum) %>% mean
pro2 <- liver_500 %>% apply(1,sum) %>% mean
@
\begin{center}
\begin{tabular}{c|c}
simulées & réelles \\
\hline
\Sexpr{pro1}
&
\Sexpr{pro2}
\end{tabular}
\end{center}
\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}{Annexe}{Opération de base}
<<fig.height=4>>=
data("Hongite")
h <- Hongite[,1:3] %>% closure
ternary_diagram_vide()
h %>% coord_ternary_diagram() %>% points()
center_data(h) %>% coord_ternary_diagram() %>% points(col="red", pch=3)
apply(h, 2, mean) %>% closure %>% coord_ternary_diagram() %>% points(col="green", pch=3)
legend("topright", pch=c(3,3), col=c("red", "green", "black"), legend=c("geometrique", "arithmetique"))
@
\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[allowframebreaks]{Références}
\nocite{Aitchison86}
\nocite{Capushe}
\nocite{Metagenomique}
\nocite{PPCA}
\nocite{Lecture}
\bibliographystyle{plain}
\bibliography{bibli.bib}
\end{frame}


\end{document}
